In [13]:
#Import Libraries
import pandas as pd
import numpy as np
import scipy.stats as stats

In [14]:
#Create DataFrame
df = pd.DataFrame ({"TransactionAmount":[100,150,50,100,130,120,100,85,70,150,150,120,50,100,100,140,90,150,50,90,120,100,110,75,65]})
df

TransactionAmount
0                 100
1                 150
2                  50
3                 100
4                 130
5                 120
6                 100
7                  85
8                  70
9                 150
10                150
11                120
12                 50
13                100
14                100
15                140
16                 90
17                150
18                 50
19                 90
20                120
21                100
22                110
23                 75
24                 65

In [15]:
#Central Tendency
import statistics
print ("Mean dari TransactionAmount adalah: ",statistics.mean(df['TransactionAmount']))# print mean
print ("Median dari TransactionAmount adalah: ",statistics.median(df['TransactionAmount']))# print median
print ("Mode dari TransactionAmount adalah: ",statistics.mode(df['TransactionAmount']))# print mode

Mean dari TransactionAmount adalah:  102.6
Median dari TransactionAmount adalah:  100
Mode dari TransactionAmount adalah:  100


**bold text**
Analisis
1. Mean dan Median memiliki nilai yang berdekatan yaitu 102.6 dan 100, berarti persebaran data bersifat simetris
2. Mode bernilai 100, berarti mayoritas transaksi yang terjadi bernilai 100
3. Mean lebih tinggi dari Median, distribusi right skewed, berarti transaksi yang terjadi lebih banyak bernilai dibawah Mean.

In [16]:
#Variability
print ("Variance dari TransactionAmount adalah",np.var(df['TransactionAmount'])) # print var
print ("Standar Deviasi dari TransactionAmount adalah",np.std(df['TransactionAmount'])) # print std
print ("Nilai Max & Min dari TransactionAmount adalah",np.max(df['TransactionAmount']),"dan", np.min(df['TransactionAmount'])) #print max & min
print ("Nilai Quantile 0.25 dari TransactionAmount adalah",np.quantile(df['TransactionAmount'],0.25)) # print quantile 0.25
print ("Nilai Quantile 0.50 dari TransactionAmount adalah",np.quantile(df['TransactionAmount'],0.50)) # print quantile 0.5
print ("Nilai Quantile 0.75 dari TransactionAmount adalah",np.quantile(df['TransactionAmount'],0.75)) #print quantile 0.75

Variance dari TransactionAmount adalah 972.2399999999999
Standar Deviasi dari TransactionAmount adalah 31.18076330047101
Nilai Max & Min dari TransactionAmount adalah 150 dan 50
Nilai Quantile 0.25 dari TransactionAmount adalah 85.0
Nilai Quantile 0.50 dari TransactionAmount adalah 100.0
Nilai Quantile 0.75 dari TransactionAmount adalah 120.0


Analisis
1. Memiliki variance yang tinggi yaitu 972.2399999999999, berarti ada transaksi yang jauh lebih besar ataupun jauh lebih kecil dari nilai rata-rata
2. Rentang nilai transaksi cukup lebar yaitu dari 50 hingga 150
3. Standar deviasi dari nilai Mean adalah 31.18076330047101. Nilai ini menunjukkan seberapa tersebar data TransactionAmount dari Mean

In [17]:
#Define H0 and H1
#H0: Rata-rata nilai transaksi = $100
#H1: Rata-rata nilai transaksi  > $100

In [18]:
#Define alpha = 0.05

In [19]:
#Calculate t statistics and P-value
#Perform one sample t-test
stats.ttest_1samp(a=df, popmean = 100)

TtestResult(statistic=array([0.40850016]), pvalue=array([0.68652848]), df=array([24]))

In [20]:
#Ttest = 0.40850016, pvalue = 0.68652848

In [21]:
#Conclude with alpha and p-value
#p-value hasil test lebih besar dari alpha (5%), maka, gagal untuk menolak H0
#Conclusion: Training kepada 25 sampel pekerja sales tidak efektif, rata-rata transaksi masih = $100

In [22]:
#Conclude with t statistics and critical region
#Find the critical region

In [23]:
#H1: Rata-rata nilai transaksi  > $100, kearah kanan
stats.t.ppf(1-0.05,24)

1.7108820799094275

In [24]:
#Critical region result
#Critical region: t > 1.7108820799094275, sedangkan t-statistics = 0.40850016.
#Berarti t-statistics berada diluar critical region atau tidak ada di critical region
#dengan kata lain, berdasarkan perhitungan maka dinyatakan gagal menolak H0
#Conclusion: Training kepada 25 sampel pekerja sales tidak efektif, rata-rata transaksi masih = $100